src/main/java/com/es/phoneshop/model/search/SearchFilter.java

```diff
@@ -0,0 +1,34 @@
+package com.es.phoneshop.model.search;
+
+import com.es.phoneshop.model.product.Product;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Locale;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class SearchFilter {
+
+    private final Object LOCK = new Object();
+    private Function<Product, Long> countMatchesProductsFunction;
```

Comment:
Why do we need to declare this variable as class field instead of local variable?